src/community/features-templating/features-templating-core/src/main/java/org/geoserver/featurestemplating/builders/flat/AttributeNameHelper.java

```diff
@@ -21,9 +22,9 @@
         this.separator = separator;
     }
 
-    String getFinalAttributeName() {
+    String getFinalAttributeName(TemplateBuilderContext context) {
         String parentKey = this.parentKey;
-        String key = this.key != null ? this.key.evaluate(null).toString() : null;
+        String key = this.key != null ? this.key.evaluate(context).toString() : null;
```

Comment:
I think this key.evaluate(context) should be evaluated on context.getCurrentObject() instead